doc/vxlan/Vxlan_hld.md

```diff
@@ -0,0 +1,391 @@
+| 0.1 |             |     Prince Sunny   | Initial version                   |
+
+# About this Manual
+This document provides general information about the Vxlan feature implementation in SONiC.
+# Scope
+This document describes the high level design of the Vxlan feature. Kernel VRF (L3mdev) programming for VNET peering is beyond the scope of this document. 
+
+# Definitions/Abbreviation
+###### Table 1: Abbreviations
+|                          |                                |
+|--------------------------|--------------------------------|
+| VNI                      | Vxlan Network Identifier       |
+| VTEP                     | Vxlan Tunnel End Point         |
+| VM                       | Virtual Machine                |
+| VRF                      | Virtual Routing and Forwarding |
+| VNet                     | Virtual Network                |
+
+# 1 Requirements Overview
+## 1.1 Functional requirements
+This section describes the SONiC requirements for Vxlan feature. 
+
+At a high level the following should be supported:
+
+Phase #1
+- Should be able to perform the role of Vxlan Tunnel End Point (VTEP)
+- VNet peering between customer VMs and Baremetal servers [VNet Requirements](https://github.com/lguohan/SAI/blob/vni/doc/SAI-Proposal-QinQ-VXLAN.md).
+- Distributed Vxlan routing with Symmetric IRB model (RIOT)
+
+Phase #2
+- Integration with BGP EVPN 
+- Should support untagged or tagged traffic (Overlay layer 2 networks over layer 3 underlay)
+- Should be able to do HER for unicast traffic based on configured flood list
+- CLI commands to configure Vxlan
+
+
+## 1.2 Orchagent requirements
+### Vxlan orchagent:
+ - Should be able to create VRF/VLAN to VNI mapping.
+ - Should be able to create NH Tunnel and Tunnel termination tables.  
+ - Should be able to create tunnels and encap/decap mappers. 
+
+### Vrf orchagent:
+ - Should be able to create VRFs per VNET tables. 
+ - Should be able to track peering configurations.
+ - Should be VNet/VRF aware
+ 
+### Route orchagent:
+ - Should be able to handle routes within a VNet 
+ - Should be able to create NH tunnels for the endpoints
+ - Should be VNet/VRF aware
+
+### FDB orchagent:
+ - Should be VTEP aware
+ - Should support static configuration of FDB entries learnt on remote VTEP
+ 
+### INTFs orchagent:
+ - Should be VRF aware
+ - Should be able to create router interfaces in a specific VRF
+ 
+## 1.3 CLI requirements
+- User should be able to get FDB learnt per VNI
+- User should be able to configure Vxlan tunnels and VTEPs (Overlay)
+
+In summary:
+```
+	- config vxlan <vxlan_name> vlan <vlan_id> vni <vni_id>
+	- config vxlan <vxlan_name> src_if <interface>
+	- config vxlan <vxlan_name> vlan <vlan_id> flood vtep <ip1, ip2, ip3>
+	- show mac vxlan <vxlan_name> <vni_id>
+	- show vxlan <vxlan_name>
+```
+Configuring VNet peering via CLI is beyond the scope
+
+## 1.4 Scalability requirements
+
+### 2.1.1 VNet Peering
+###### Table 2: VNet peering scalability
```

Comment:
What will be the total number of routes to be supported ? Can that be added to this table as well. - thx